From 71c770aea48c6be7228ae4051b8d875f953f52d7 Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Thu, 16 Jan 2020 11:12:36 -0800
Subject: Rename smime oppenc mode parameter to get_keys_by_addr()

Make it the same as with pgpkey.c and crypt-gpgme.c, to remove
confusion in the future about this (as I had briefly).

This is also needed because the
$crypt_opportunistic_encrypt_strong_keys logic added to smime.c relies
on the "may_ask" parameter being a reflection of oppenc mode, not just
a "quiet" mode.
---
 smime.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/smime.c b/smime.c
index 90d6769b..0ee56535 100644
--- a/smime.c
+++ b/smime.c
@@ -658,7 +658,7 @@ static smime_key_t *smime_get_key_by_hash(char *hash, short public)
   return match;
 }
 
-static smime_key_t *smime_get_key_by_addr(char *mailbox, short abilities, short public, short may_ask)
+static smime_key_t *smime_get_key_by_addr(char *mailbox, short abilities, short public, short oppenc_mode)
 {
   smime_key_t *results, *result;
   smime_key_t *matches = NULL;
@@ -706,7 +706,7 @@ static smime_key_t *smime_get_key_by_addr(char *mailbox, short abilities, short
 
   if (matches)
   {
-    if (! may_ask)
+    if (oppenc_mode)
     {
       if (trusted_match)
         return_key = smime_copy_key (trusted_match);
@@ -807,7 +807,7 @@ void _smime_getkeys (char *mailbox)
   char *k = NULL;
   char buf[STRING];
 
-  key = smime_get_key_by_addr (mailbox, KEYFLAG_CANENCRYPT, 0, 1);
+  key = smime_get_key_by_addr (mailbox, KEYFLAG_CANENCRYPT, 0, 0);
 
   if (!key)
   {
@@ -913,7 +913,7 @@ char *smime_findKeys (ADDRESS *adrlist, int oppenc_mode)
 
     q = p;
 
-    key = smime_get_key_by_addr (q->mailbox, KEYFLAG_CANENCRYPT, 1, !oppenc_mode);
+    key = smime_get_key_by_addr (q->mailbox, KEYFLAG_CANENCRYPT, 1, oppenc_mode);
     if ((key == NULL) && (! oppenc_mode))
     {
       snprintf(buf, sizeof(buf),
